### DISCUSSION TITLE
PAP backend for CUPS to share AppleTalk printer to modern systems

### DISCUSSION THREAD

**Discussion by mabam** [Netatalk/netatalk#1656] [Ideas]
A couple of years ago I wrote a guide on how to build an AppleTalk NAS using A2SERVER. It contains a section explaining how to share a LocalTalk LaserWriter to modern systems:
https://www.emaculation.com/doku.php/appletalk_printserver_macos_and_osx#optional_set_up_the_print_server.

It also points to a download link of a fixed PAP backend to be used with CUPS. Thought this might be interesting for https://netatalk.io/links and/or https://github.com/Netatalk/netatalk/wiki/Print-Server.

**Comment by NJRoadfan** [comment]
This how-to covers printer setup as well: https://biosrhythm.com/?p=2791

Keep in mind that printer backends will be obsolete and unsupported with CUPS 3.0. One would have to create a "printer application" to do that in the future.

**Comment by rdmark** [comment]
[converted to a Discussion]

@mabam Thanks for sharing these resources. I think this is a great fit for the Print Server wiki page. This project has been lacking documentation / guides on how to print to LocalTalk printers for a long time. Mainly because I don't own such a printer myself. If you have a moment to spare, please go ahead and edit the wiki page yourself. You can add a heading for "Printing to LocalTalk printers" and put a link to your resources there, with whatever context you think is appropriate. I'm looking forward to your contribution. :)
